tools/wpt/browser.py

```diff
@@ -537,7 +537,7 @@ def _get_latest_chromium_revision(self, architecture):
                         f"{architecture}/LAST_CHANGE")
         return get(revision_url).text.strip()
 
-    def _get_chromium_download_url(self, version=None):
+    def _get_chromium_download_url(self, filename, version=None):
```

Comment:
I do agree with this. This strange approach was due to saving the successfully used url to ensure we install any further components from the same URL (but saving the url without the filename). I've done some refactoring so that we pass around the revision and save the revision used if it was successful, and added a helper method to build the snapshots URL string by passing in all the components.